apps/laboratory/tests/smart-account.spec.ts

```diff
@@ -45,8 +47,26 @@ testModalSmartAccount(
     const walletModalValidator = modalValidator as ModalWalletValidator
 
     await walletModalPage.togglePreferredAccountType()
-    await walletModalPage.switchNetwork('Polygon')
+    await walletModalPage.switchNetwork('Avalanche')
     await walletModalPage.openSettings()
     await walletModalValidator.expectEoaAddress(testInfo.parallelIndex)
   }
 )
+
+testModalSmartAccount(
+  'it should use an eoa when disconnecting and connecting to a not enabled address',
+  async ({ modalPage, modalValidator, context }) => {
+    const walletModalPage = modalPage as ModalWalletPage
+    const walletModalValidator = modalValidator as ModalWalletValidator
+
+    await walletModalPage.togglePreferredAccountType()
+    await walletModalPage.disconnect()
+    await walletModalPage.page.waitForTimeout(2500)
```

Comment:
Is this absolutely required?

Reply:
it was preventing the magic disconnect promise from resolving, but shoould work without it.
will push fix in a diff pr if possible